### PR TITLE
Add getrole endpoint

### PR DESCRIPTION
**Pull request by KanashinDmitry**

### CODE REVIEW

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LT.DigitalOffice.ProjectService.Mappers.ResponsesMappers.Interfaces
+{
+    public interface IRoleExpandedResponseMapper
+    {
+        RoleExpandedResponse Map(DbRole dbRole);
+    }
```
Comment (KostinP) [inline src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs:12]: Why not IMapper<DbRole, RoleExpandedResponse>?

File: src/ProjectService.Models.Dto/Responses/RoleExpandedResponse.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using LT.DigitalOffice.ProjectService.Models.Dto.ResponsesModels;
+
+namespace LT.DigitalOffice.ProjectService.Models.Dto.Responses
+{
+    public class RoleExpandedResponse
+    {
+        public Role Role { get; set; }
+    }
```
Comment (KostinP) [inline src/ProjectService.Models.Dto/Responses/RoleExpandedResponse.cs:8]: Why create a new model?

File: src/ProjectService.Models.Dto/Responses/RolesResponse.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using LT.DigitalOffice.ProjectService.Models.Dto.ResponsesModels;
+using System.Collections.Generic;
+
+namespace LT.DigitalOffice.ProjectService.Models.Dto.Responses
+{
+    public class RolesResponse
+    {
+        public IEnumerable<Role> Roles { get; set; }
+        public int TotalCount { get; set; }
```
Comment (KostinP) [inline src/ProjectService.Models.Dto/Responses/RolesResponse.cs:9]: Why is the total number needed?

File: src/ProjectService/Controllers/RoleController.cs [hunk 4]
```diff
@@ -1,19 +1,36 @@
...
     public class RoleController : ControllerBase
     {
+        [HttpGet("getUserRoleInProject")]
+        public RoleExpandedResponse GetUserRoleInProject(
+            [FromServices] IGetRoleCommand command,
+            [FromQuery] Guid roleId)
+        {
+            return command.Execute(roleId);
+        }
         [HttpDelete("disableUserRoleInProject")]
```
Comment (KostinP) [inline src/ProjectService/Controllers/RoleController.cs:19]: Add empty line

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+using LT.DigitalOffice.ProjectService.Models.Dto.Responses;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace LT.DigitalOffice.ProjectService.Mappers.ResponsesMappers.Interfaces
+{
+    public interface IRolesResponseMapper
+    {
+        RolesResponse Map(IEnumerable<DbRole> dbRoles);
```
Comment (djino97) [inline src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs:12]: need interface IMapper

File: src/ProjectService.Mappers/ResponsesMappers/RolesResponseMapper.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+        private readonly IRoleMapper _roleMapper;
+
+        public RolesResponseMapper(
+            IRoleMapper roleMapper)
+        {
+            _roleMapper = roleMapper;
+        }
+
+        public RolesResponse Map(IEnumerable<DbRole> dbRoles)
+        {
```
Comment (djino97) [inline src/ProjectService.Mappers/ResponsesMappers/RolesResponseMapper.cs:21]: Need check dbRoles at null

Comment (KanashinDmitry) [reply src/ProjectService.Mappers/ResponsesMappers/RolesResponseMapper.cs:22]: Added

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs (hunk 5 above)
Comment (KanashinDmitry) [reply src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs:12]: Added

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs (hunk 1 above)
Comment (KanashinDmitry) [reply src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs]: Forgot about it, now added

File: src/ProjectService/Controllers/RoleController.cs (hunk 4 above)
Comment (KanashinDmitry) [reply src/ProjectService/Controllers/RoleController.cs]: Added

File: src/ProjectService.Models.Dto/Responses/RolesResponse.cs (hunk 3 above)
Comment (KanashinDmitry) [reply src/ProjectService.Models.Dto/Responses/RolesResponse.cs:9]: It is needed for the frontend to make pagination (to make sure how many roles they can show and skip)

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs (hunk 5 above)
Comment (KostinP) [reply src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRolesResponseMapper.cs:12]: Your `IRolesResponseMapper` interface now inherits from the `IMapper` interface, which contains a `RolesResponse Map(IEnumerable<DbRole> dbRoles);` member. So it is already in the `IRolesResponseMapper`. 

So remove `RolesResponse Map(IEnumerable<DbRole> dbRoles);`

File: src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs (hunk 1 above)
Comment (KostinP) [reply src/ProjectService.Mappers/ResponsesMappers/Interfaces/IRoleExpandedResponseMapper.cs]: Explained in a discussion with @djino97 what was wrong

**Comment by djino97** [comment]
**This is pull request need move in UserService repository**

**Pull request closed without merging**